lib/core/message.toit

```diff
@@ -88,23 +93,82 @@ process_messages_:
       // task's deadline if any.
       critical_do --no-respect_deadline:
         if message is Array_:
-          type ::= message[0]
-          handler ::= system_message_handlers_.get type --if_absent=:
-            print_ "WARNING: unhandled system message $type"
-            continue
-          gid ::= message[1]
-          pid ::= message[2]
-          arguments ::= message[3]
-          message = null  // Allow garbage collector to free.
-
-          handler.on_message type gid pid arguments
+          entered := false
+          lambda := ::
+            entered = true
+            type ::= message[0]
+            handler ::= system_message_handlers_.get type
+            if handler:
+              gid ::= message[1]
+              pid ::= message[2]
+              arguments ::= message[3]
+              message = null  // Allow garbage collector to free.
```

Comment:
Hopefully shouldn't be necessary in the future when the variable slots are reused.
Also not sure it buys a lot, but can't really hurt.

Reply:
Dropped this in another change.